### DISCUSSION TITLE
Auto increment id increments by a random(?) number

### DISCUSSION THREAD

**Discussion by shawshankkumar** [supabase/supabase#11711] [Questions] [answered]
I recently made a simple registration form for a college community event and used supabase as the backend. I noticed that a few times the auto increment id in consecutive rows differ by 3 or 4. Why does this happen?

**Comment by GaryAustin1** [comment, marked as answer]
It gets incremented even when the operation fails, or does not meet RLS.
Lots of hits on this for postgres on the web.

**Reply by shawshankkumar** [reply]
> It gets incremented even when the operation fails, or does not meet RLS. Lots of hits on this for postgres on the web.

Oh, thanks for the information, apologies I didn't know this, new to postgres!
